samcli/commands/local/cli_common/invoke_context.py

```diff
@@ -219,6 +225,10 @@ def __enter__(self) -> "InvokeContext":
 
         self._stacks = self._get_stacks()
 
+        if self._ctx:
+            LOG.info("Validating template %s\n", self._template_file)
+            run_sam_validate(ctx=self._ctx, template=self._template_file, lint=True)
```

Comment:
Is this going to fail the command if validation fails?

Reply:
seems like it could raise User Exception. Does it make sense to not be able to run invoke if a template is invalid?